site/src/api/queries/users.ts

```diff
@@ -89,21 +87,11 @@ export const authMethods = () => {
   };
 };
 
-const initialMeData = getMetadataAsJSON<User>("user");
-const meKey = ["me"] as const;
-
-export const me = (queryClient: QueryClient) => {
+export const me = () => {
   return {
-    queryKey: meKey,
-    queryFn: async () => {
-      const cachedData = queryClient.getQueryData(meKey);
-      if (cachedData === undefined && initialMeData !== undefined) {
-        return initialMeData;
-      }
-
-      return API.getAuthenticatedUser();
-    },
-  } satisfies UseQueryOptions<User>;
```

Comment:
Hm... @Parkreiner maybe we should do something different like:
```ts
 return {
    queryKey: meKey,
    initialData: initialMe,
    queryFn: API.getAuthenticatedUser,
 }
 ```
Because if the `initialData` is `undefined` the query will be loaded anyways.

Reply:
Yeah...This works perfectly. I double-checked the React Query type definitions, and somehow missed that initialData had an extra `undefined` type tagged on to its allowable values, and that it wouldn't affect the type parameters at all. I also double-checked the documentation, and [found a section on this use case](https://tanstack.com/query/v4/docs/react/guides/initial-query-data#conditional-initial-data-from-cache)
 
 I'm going to make another PR simplifying all the other functions that work like this